src/engine/strat_engine/backstore/blockdevmgr.rs

```diff
@@ -690,6 +690,21 @@ impl BlockDevMgr {
         }
     }
 
+    pub fn grow(
+        &mut self,
+        data_segments: &[(Sectors, Sectors)],
+        dev: DevUuid,
+    ) -> StratisResult<bool> {
+        let bd = self
+            .block_devs
+            .iter_mut()
+            .find(|bd| bd.uuid() == dev)
+            .ok_or_else(|| {
+                StratisError::Msg(format!("Block device with UUID {} not found", dev))
+            })?;
+        bd.grow(data_segments)
```

Comment:
We remake the `RangeAllocator` with a new size, that's the difference. Would you prefer a change to `RangeAllocator` that increases the size as opposed to creating a new one with a larger size? That might be cleaner.

Reply:
Yes. I think we could to that quite safely because the only thing that is necessary to ensure that the RangeAllocator is well formed is that the new size is greater than the old.